app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -49,18 +50,25 @@ export default function*() {
   ];
 }
 
-function* createPageSession({ id, location, options }) {
+function* createPageSession({ id, location, options, withUI = true }) {
```

Comment:
Hmm.  Looking ahead, I expect the "bulk" operations to diverge pretty rapidly from the "respond to UI action" operations.  Would it be wiser to factor this into a "replacePageSessions" bulk action, and especially to add a "destroyPageSessions" action, for when we implement "Close All Tabs" / "Close Other Tabs", etc?

Also, it looks like you're starting a new UAS session for each page you restore.  Are you sure that's what you want?  I think we should at least tell the UAS this is a session restore; seems like the kind of data we might want to use and/or surface to say, "Hey, this tab has been restored 7 times, should I snooze it?"  We might also want to give you a bulk UAS call for this, so you don't spawn 1000 UAS POSTs to session restore.

None of these necessarily block, but they're worth thinking about.